contracts/Marketplace.sol

```diff
@@ -1,51 +1,324 @@
+    // ? There is a require statement to check if the owner is the same as the one who made the listing. I'm using the onlyOwner on this function but don't know if works like that.
+    function deleteMarketItem(uint256 itemId) public onlyOwner nonReentrant {
+        if (itemId >= _itemCounter.current()) {
+            revert IdLessOrEqualItemCount();
+        }
+        if (marketItems[itemId].state != MarketItemState.Listed) {
+            revert ItemMustBeOnMarketplace();
+        }
+        // require(
+        //     itemId <= _itemCounter.current(),
+        //     "id must be less or equal than item count"
+        // );
+        // require(
+        //     marketItems[itemId].state == MarketItemState.Listed,
+        //     "Item must be on the Marketplace"
+        // );
+
+        MarketItem storage item = marketItems[itemId];
+
+        // ! Modify this require to match ERC1155 contract
+        // * We can also add a custom error here
+        // require(
+        //     IERC721(item.nftContract).ownerOf(item.tokenId) == msg.sender,
+        //     "must be the owner"
+        // );
+        // require(
+        //     IERC721(item.nftContract).getApproved(item.tokenId) ==
+        //         address(this),
+        //     "NFT must be approved to market"
+        // );
+
+        item.state = MarketItemState.Inactive;
+
+        emit MarketItemSold(
+            itemId,
+            item.nftContract,
+            item.tokenId,
+            item.seller,
+            address(0),
+            0,
+            MarketItemState.Inactive
+        );
+    }
+
+    // * BUY AN NFT ITEM
+    // * Transfer ownership of the item to the buyer, as well as founds to the seller
+    function createMarketSale(address nftContract, uint256 id)
+        public
+        payable
+        nonReentrant
+    {
+        MarketItem storage item = marketItems[id];
+        uint256 price = item.price;
+        uint256 tokenId = item.tokenId;
+
+        if (msg.value != price) {
+            revert SubmitActualPrice();
+        }
+        // ! Modify this require to match ERC1155 contract
+        // * We can also add a custom error here
+        // require(IERC721(nftContract).getApproved(tokenId) == address(this), "NFT must be approved to market");
+
+        item.buyer = payable(msg.sender);
+        item.state = MarketItemState.Release;
+        _itemSoldCounter.increment();
+
+        // ! Modify this require to match ERC1155 contract
+        // On ERC1155 there are mutiple functions to transfer like: safeTransferFrom
+        /** Params/properties for the function
+          - address from
+          - address to
+          - uint256 id
+          - uint256 amount
+          - bytes memory data
+          */
+        ERC1155(nftContract).safeTransferFrom(
+            item.seller, // ? Are those correct?
+            item.buyer, // ? Are those correct?
+            tokenId, // ? Are those correct?
+            price,
+            bytes("") // ????
```

Comment:
This is correct. The `data` parameter is for passing along extra information that we aren't (currently?) using so just leave it empty like you do.